h/form.py

```diff
@@ -15,16 +20,29 @@
 )
 
 
-def init():
-    from deform import Form
-    from deform_jinja2 import jinja2_renderer_factory
-    from deform_jinja2.translator import PyramidTranslator
+# We supply our own Jinja2 renderer factory because deform_jinja2's one doesn't
+# support autoescaping.
+def jinja2_renderer_factory(config):
+    environment = []
+
+    def jinja2_renderer(template_name, **kwargs):
+        if not environment:
+            searchpath = [pkg_resources.resource_filename(*(path.split(':')))
+                          for path in SEARCH_PATHS]
+
+            environment.append(config.get_jinja2_environment().overlay(
+                loader=pyramid_jinja2.SmartAssetSpecLoader(
+                    searchpath=searchpath)))
+
+        if '.jinja2' not in template_name:
+            template_name += '.jinja2'
+
+        template = environment[0].get_template(template_name)
```

Comment:
I was a bit confused by why you're storing the environment in a list... but I guess you're working around the fact that the jinja2 environment might not exist when `includeme` is called.

It would be more conventional to use `config.action` much as it's used in `app.py` to ensure that this is set up after the configuration stage of the application boot.